pkg/ingester/index/index.go

```diff
@@ -244,6 +244,11 @@ func (shard *indexShard) delete(labels labels.Labels, fp model.Fingerprint) {
 		j := sort.Search(len(fingerprints.fps), func(i int) bool {
 			return fingerprints.fps[i] >= fp
 		})
+
+		// see if search didn't find fp which matches the condition which means we don't have to do anything.
+		if j == len(fingerprints.fps) {
```

Comment:
What if the desired fingerprint is not in the slice, but there are other fingerprints with a greater value?
The search will return the position in the slice where the fp should have been.  
eg if `fingerprints.fps=[0,1,3]` and you search for the fingerprint with a value of "2", `j` will be set to 2.  This will then result in the fingerprint with value of "3" being deleted.

Looks like the checks used in the docs, https://golang.org/pkg/sort/#Search are probably what is needed here.

Reply:
That's a valid point, we need to verify that returned value is `< len(fps)` and value at returned index is what we're looking for.

Reply:
I have fixed it. Thanks for the feedback!